packages/stack/src/views/Stack/CardStack.tsx

```diff
@@ -667,7 +690,7 @@ export class CardStack extends React.Component<Props, State> {
             return (
               <MaybeScreen
                 key={route.key}
-                style={StyleSheet.absoluteFill}
+                style={[StyleSheet.absoluteFill, { zIndex: index }]}
```

Comment:
I don't know if this is correct, but I believe that if keys are unchanged, those re not getting re-ordered. At leat, on iOS.